astropy/coordinates/builtin_frames/skyoffset.py

```diff
@@ -43,36 +43,17 @@ def make_skyoffset_cls(framecls):
     if framecls in _skyoffset_cache:
         return _skyoffset_cache[framecls]
 
-    # the class of a class object is the metaclass
-    framemeta = framecls.__class__
-
-    class SkyOffsetMeta(framemeta):
-        """
-        This metaclass renames the class to be "SkyOffset<framecls>" and also
-        adjusts the frame specific representation info so that spherical names
-        are always "lon" and "lat" (instead of e.g. "ra" and "dec").
-        """
-
-        def __new__(cls, name, bases, members):
-            # Only 'origin' is needed here, to set the origin frame properly.
-            members['origin'] = CoordinateAttribute(frame=framecls, default=None)
-
-            # This has to be done because FrameMeta will set these attributes
-            # to the defaults from BaseCoordinateFrame when it creates the base
-            # SkyOffsetFrame class initially.
-            members['_default_representation'] = framecls._default_representation
-            members['_default_differential'] = framecls._default_differential
-
-            newname = name[:-5] if name.endswith('Frame') else name
-            newname += framecls.__name__
-
-            return super().__new__(cls, newname, bases, members)
-
-    # We need this to handle the intermediate metaclass correctly, otherwise we could
-    # just subclass SkyOffsetFrame.
-    _SkyOffsetFramecls = SkyOffsetMeta('SkyOffsetFrame',
-                                       (SkyOffsetFrame, framecls),
-                                       {'__doc__': SkyOffsetFrame.__doc__})
+    # Create a new SkyOffsetFrame subclass for this frame class.
+    name = 'SkyOffset' + framecls.__name__
+    _SkyOffsetFramecls = type(
+        name, (SkyOffsetFrame, framecls),
+        {'origin': CoordinateAttribute(frame=framecls, default=None),
+         # The following two have to be done because otherwise we use the
+         # defaults of SkyOffsetFrame set by BaseCoordinateFrame.
+         '_default_representation': framecls._default_representation,
+         '_default_differential': framecls._default_differential,
+         '__doc__': SkyOffsetFrame.__doc__,
+         })
```

Comment:
:+1: this is simpler.

Reply:
I had actually hoped to somehow use `__init_subclass__` for this, but it seems at that point it is too late to define a descriptor on the class (it doesn't seem to get bound). Anyway, this is fairly clean.